includes/Modules/Sign_In_With_Google.php

```diff
@@ -53,52 +66,208 @@ final class Sign_In_With_Google extends Module implements Module_With_Assets, Mo
 	public function register() {
 		add_filter( 'wp_login_errors', array( $this, 'handle_google_auth_errors' ) );
 
-		add_action( 'login_form_google_auth', array( $this, 'handle_google_auth' ) );
+		add_action( 'login_form_google_auth', $this->get_method_proxy( 'handle_auth_callback' ) );
 		add_action( 'login_form', $this->get_method_proxy( 'render_signin_button' ) );
 	}
 
 	/**
-	 * Intercept the page request to process token ID
-	 * and complete Sign in with Google flow.
+	 * Handles the callback request after the user signs in with Google.
 	 *
 	 * @since n.e.x.t
 	 */
-	public function handle_google_auth() {
+	private function handle_auth_callback() {
```

Comment:
Should we maybe extract the auth behavior to its own class?

Reply:
Added main tests for non-multisite cases. Let me know if it looks good to you and I'll check to add ms tests as well.